src/features/contacts/index.tsx

```diff
@@ -1,22 +1,37 @@
-import { Avatar, ListItemAvatar } from "@mui/material";
+import { Avatar, Button, ListItemAvatar } from "@mui/material";
 import { ContactDetailsDto, ContactImportDto } from "lib/network/swagger-client";
 import { useRequestContext } from "providers/request-provider";
 import { ContactNameListItem, ContactNameListItemText } from "./index.styled";
 import {
+  contactGridSettingsStorageKey,
   contactListPageBreadcrumb,
   defaultFilterOrderColumn,
   defaultFilterOrderDirection,
   modelName,
   searchLabel,
 } from "./constants";
-import { DataList } from "components/data-list";
+import { DataList } from "@components/data-list";
 import { GridColDef } from "@mui/x-data-grid";
-import { CoreModule } from "lib/router";
+import { CoreModule, getAddFormRoute } from "lib/router";
 import { dataListBreadcrumbLinks } from "utils/constants";
+import { ModuleWrapper } from "@components/module-wrapper";
+import { SearchBar } from "@components/search-bar";
+import { useEffect, useState } from "react";
+import { Add, Download, Upload } from "@mui/icons-material";
+import { GhostLink } from "@components/ghost-link";
+import { getLocalStorageSavedPropertyValue } from "utils/helper";
 
 export const Contacts = () => {
   const { client } = useRequestContext();
 
+  const [searchTerm, setSearchTerm] = useState("");
+  const [openImport, setOpenImport] = useState(false);
+  const [openExport, setOpenExport] = useState(false);
+
+  useEffect(() => {
+    setSearchTerm(getLocalStorageSavedPropertyValue(contactGridSettingsStorageKey, "searchTerm"));
```

Comment:
why do you save search and filter state?
for me it looks as if google opening with last search and not ready to use for search

Reply:
As I understand it was Peter who is wanted this feature like this

Reply:
Yeah. Its a requirement to preserve last search & filter state of the grid.